### PR TITLE
Kim-Linh Vu - Hack4Impact Milestone #3

### PR DESCRIPTION
**Pull request by KimLinhVu**
# Milestone: Milestone #3
# Developer: Kim-Linh Vu

### Pull Request Summary

This is the completed milestone 3 being submitted for review.

### Pull Request Checklist
- [x] Code is neat, readable, and works
- [x] Comments are appropriate
- [x] The commit message follows our guidelines
- [x] The milestone number is specified
- [x] The developer name is specified
- [x] The summary is completed

### Reviewer Checklist  - PULL REQUEST REVIEWER ONLY

IMPORTANT: The rest of the sections in this checklist should only be filled out by authorized pull request reviewers. If you are the individual template contributor, do not fill out the rest of the fields or check the boxes.

NOTE: Milestones can only be completed when all boxes are checked.

- [ ] The code is fully reviewed
- [ ] Meaningful feedback is given
- [ ] Comment that you have reviewed the code
- [ ] This box is checked

### CODE REVIEW

File: milestone3/frontend/src/recipeData.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    instructions: ["Start by preparing the sauce. Add 3 tablespoons sesame paste to a bowl, and gradually stir in 2 tablespoons sesame oil to smooth out the paste.", "Then add 1 ½ teaspoons dark soy sauce and 1 tablespoon light soy sauce (or 4 ½ teaspoons of master sauce, if you have it). Add 1 teaspoon sugar, and stir until well combined. If the sauce is still too thick, add in a few drops of warm water or more master sauce to thin it out. Add salt to taste and set aside. You can prepare this sauce ahead of time.", "Boil a pot of water to cook the noodle as per package instructions, undercooking it by 1 minute to ensure they remain chewy and al dente. Drain. Shake off any excess liquid from the noodles, and put them in a large bowl.", "Toss the noodles with the remaining 1 ½ teaspoons sesame oil to prevent noodles from sticking together. Add in the preserved radish, pickled long beans, garlic, chili oil, cilantro, scallions, vinegar, and tahini sauce. Quickly toss the noodles together and serve hot."]}
```
Comment (sfwathen) [inline milestone3/frontend/src/recipeData.ts]: Hi Kim-Linh, 
Could you resubmit this milestone using your own recipes? (not the ones from the example site). If you are struggling to think of some you are welcome to look recipes up and summarize from there. Sorry we didn't catch this in milestone 1.

File: milestone3/frontend/src/components/RecipePage.tsx [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+}
+
+export default function RecipePage(props: RecipePageProps) {
+    const { name, id } = useParams();
+    const defaultRecipe = recipes[0];
+    const [recipe, setRecipe] = useState(defaultRecipe)
+    const [newIngredient, setNewIngredient] = useState('');
+    const [allIngredients, setAllIngredients] = useState(defaultRecipe.ingredients)
+    const [newInstruction, setNewInstruction] = useState('');
+    const [allInstructions, setAllInstructions] = useState(defaultRecipe.instructions)
```
Comment (sfwathen) [inline milestone3/frontend/src/components/RecipePage.tsx]: So we don't want to set our ingredients and instructions to be from the defaultRecipe. If you look at your pages, this is causing all of your recipes to have the same ingredients and instructions list as the first recipe! 
To fix this, you can call the setAllIngredients and setAllInstructions functions with the recipe we get back in our useEffect. This way the instructions and ingredients lists will reflect the current recipe on this page

File: milestone3/frontend/src/components/RecipePage.tsx [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+    useEffect(() => {
+        if (props.external) {
+            fetch(`https://bootcamp-milestone-4.onrender.com/recipe/${name}`)
+                .then(res => res.json())
+                .then(data => {setRecipe(data[0])})
```
Comment (sfwathen) [inline milestone3/frontend/src/components/RecipePage.tsx]: Try to add a .catch() to the end of your .then() chains so you can catch and print any errors!

**Review by sfwathen — CHANGES_REQUESTED** [review]
Hi Kim-Linh,
Could I have you resubmit this milestone with your own recipes and with the correct ingredients and instructions on each page? Sorry we didn't catch it on milestone 1 but we want this site to hold unique recipes! not the ones copied from the example site. Also, make sure to update your ingredients and instructions lists from the recipe on the current page, otherwise all your pages will have the same instructions and ingredients. 
Thanks!

File: milestone3/frontend/src/recipeData.ts (hunk 1 above)
Comment (KimLinhVu) [reply milestone3/frontend/src/recipeData.ts]: Hello, I already completed Milestone 5, so I no longer have this file in my current code (I am now pulling the recipe data from mongodb). Can I just make a pull request for Milestone 5 for review?

Comment (sfwathen) [reply milestone3/frontend/src/recipeData.ts]: Hi, yes if you have Milestone 5 done you can submit that, just make sure that you have different recipes in your database once you submit, and that the instructions and ingredients are showing up correctly on each page as I mentioned in the other comments. I will close this now so you can make a new request.

**Pull request closed without merging**